packages/runtime/src/next/components/head-tags.tsx

```diff
@@ -0,0 +1,66 @@
+import { ComponentPropsWithoutRef } from 'react'
+import { useIsPagesRouter } from '../../components/hooks/useIsPagesRouter'
+import Head from 'next/head'
+
+export function PageTitle({ children, ...props }: ComponentPropsWithoutRef<'title'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <title {...props}>{children}</title>
+      </Head>
+    )
+  }
+  return <title {...props}>{children}</title>
+}
+
+export function PageMeta(props: ComponentPropsWithoutRef<'meta'>) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <meta {...props} />
+      </Head>
+    )
+  }
+  return <meta {...props} />
+}
+
+// https://react.dev/reference/react-dom/components/link#props
+type ReactLinkProps = {
+  precedence?: 'reset' | 'low' | 'medium' | 'high'
+}
+type PageLinkProps = ComponentPropsWithoutRef<'link'> & ReactLinkProps
+
+export function PageLink({ precedence, ...props }: PageLinkProps) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <link {...props} />
+      </Head>
+    )
+  }
+  //@ts-expect-error
+  return <link {...props} precedence={precedence} />
+}
+
+// https://react.dev/reference/react-dom/components/style#props
+type ReactStyleProps = {
+  precedence?: 'reset' | 'low' | 'medium' | 'high'
+  href?: string
+}
+type PageStyleProps = ComponentPropsWithoutRef<'style'> & ReactStyleProps
+
+export function PageStyle({ children, precedence, href, ...props }: PageStyleProps) {
+  if (useIsPagesRouter()) {
+    return (
+      <Head>
+        <style {...props}>{children}</style>
```

Comment:
Is the `href` prop not a regular prop that we'd want to pass here too?

Reply:
href is not a standard attribute for `style`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style

Reply:
Ah, got it. Thanks for clarifying.